### PR TITLE
Add java OIDC docs

### PR DESCRIPTION
**Pull request by dirkkul**
Adds OIDC authentication documentation for java client

### CODE REVIEW

File: developers/weaviate/client-libraries/java.md [hunk 1]
```diff
@@ -47,22 +47,87 @@ public class App {
...
+import technology.semi.weaviate.client.Config;
+import technology.semi.weaviate.client.WeaviateAuthClient;
+
+Config config = new Config("http", "weaviate.example.com:8080");
+WeaviateAuthClient.clientPassword(
+    config=config,
+    username="Your user",
+    password="Your password",
+    scopes=Arrays.asList("offline_access") // optional, depends on the configuration of your identity provider (not required with WCS)
+);
```
Comment (aliszka) [inline developers/weaviate/client-libraries/java.md:76]:
```
WeaviateAuthClient.clientPassword(
    config,
    "Your user",
    "Your password",
    Arrays.asList("offline_access") // optional, depends on the configuration of your identity provider (not required with WCS)
);
```

File: developers/weaviate/client-libraries/java.md [hunk 2]
```diff
@@ -47,22 +47,87 @@ public class App {
...
+```java
+import technology.semi.weaviate.client.Config;
+import technology.semi.weaviate.client.WeaviateAuthClient;
+
+Config config = new Config("http", "weaviate.example.com:8080");
+WeaviateAuthClient.clientCredentials(
+    config=config,
+    clientSecret="your_client_secret",
+    scopes=Arrays.asList("scope1 scope2") // optional, depends on the configuration of your identity provider
+);
```
Comment (aliszka) [inline developers/weaviate/client-libraries/java.md:94]:
```
WeaviateAuthClient.clientCredentials(
    config,
    "your_client_secret",
    Arrays.asList("scope1", "scope2") // optional, depends on the configuration of your identity provider
);
```

File: developers/weaviate/client-libraries/java.md [hunk 3]
```diff
@@ -47,22 +47,87 @@ public class App {
...
+import technology.semi.weaviate.client.Config;
+import technology.semi.weaviate.client.WeaviateAuthClient;
+
+Config config = new Config("http", "weaviate.example.com:8080");
+WeaviateAuthClient.bearerToken(
+    config=config,
+    accessToken="Your_access_token",
+    accessTokenLifetime=500,  // in seconds
+    refreshToken="Your_refresh_token",
+);
```
Comment (aliszka) [inline developers/weaviate/client-libraries/java.md:113]:
```
WeaviateAuthClient.bearerToken(
    config,
    "Your_access_token",
    500,  // in seconds
    "Your_refresh_token"
);
```

File: developers/weaviate/client-libraries/java.md [hunk 4]
```diff
@@ -47,22 +47,87 @@ public class App {
...
+
+```java
+import technology.semi.weaviate.client.Config;
+import technology.semi.weaviate.client.WeaviateAuthClient;
+
+Config config = new Config("http", "weaviate.example.com:8080");
+WeaviateAuthClient.clientCredentials(
+    config,
+    "your_client_secret",
+    Arrays.asList("scope1,scope2") // optional, depends on the configuration of your identity provider
```
Comment (aliszka) [inline developers/weaviate/client-libraries/java.md:93]: `Arrays.asList("scope1", "scope2")`

**Review by databyjp — APPROVED** [review]

**Pull request merged**